odiglet/pkg/kube/runtime_details/inspection.go

```diff
@@ -176,7 +183,91 @@ func runtimeInspection(pods []corev1.Pod, ignoredContainers []string) ([]odigosv
+	fetchAndSetEnvVarsFromDockerfile(ctx, criClient, pod, container, envVarNames, resultsMap)
+}
+
+// fetchAndSetEnvVarsFromDockerfile retrieves environment variables from the container's Dockerfile and updates the runtime details.
+func fetchAndSetEnvVarsFromDockerfile(ctx context.Context, criClient criwrapper.CriClient, pod corev1.Pod, container corev1.Container,
+	envVarKeys []string, resultsMap *map[string]odigosv1.RuntimeDetailsByContainer) {
+	containerID := getContainerID(pod.Status.ContainerStatuses, container.Name)
+	if containerID == "" {
+		log.Logger.V(0).Info("containerID not found for container", "container", container.Name, "pod", pod.Name, "namespace", pod.Namespace)
+		return
+	}
+
+	envVars, err := criClient.GetContainerEnvVarsList(ctx, envVarKeys, containerID)
+	runtimeDetailsByContainer := (*resultsMap)[container.Name]
+
+	if err != nil {
+		log.Logger.Error(err, "failed to get relevant env var per language", "container", container.Name, "pod", pod.Name, "namespace", pod.Namespace)
+		errMessage := fmt.Sprintf("CRI communication error for container %s in pod %s/%s",
+			container.Name, pod.Namespace, pod.Name)
+		runtimeDetailsByContainer.CriErrorMessage = &errMessage
+	} else if envVars != nil {
+		runtimeDetailsByContainer.CriErrorMessage = nil
+		runtimeDetailsByContainer.EnvVarsFromDockerFile = envVars
+	}
+
+	// Update the results map with the modified runtime details
+	(*resultsMap)[container.Name] = runtimeDetailsByContainer
+}
+
+// getContainerID retrieves the container ID for a given container name from the list of container statuses.
+func getContainerID(containerStatuses []corev1.ContainerStatus, containerName string) string {
+	for _, containerStatus := range containerStatuses {
+		if containerStatus.Name == containerName {
+			return containerStatus.ContainerID
+		}
+	}
+	return ""
+}
+
+func checkEnvVarsInContainerManifest(container corev1.Container, envVarNames []string) bool {
+	// Create a map for quick lookup of envVar names
+	envVarSet := make(map[string]struct{})
+	for _, name := range envVarNames {
+		envVarSet[name] = struct{}{}
+	}
+
+	// Iterate over the container's environment variables
+	for _, containerEnvVar := range container.Env {
+		if _, exists := envVarSet[containerEnvVar.Name]; exists {
+			return true
+		}
+	}
+	return false
+}
+
 func persistRuntimeDetailsToInstrumentationConfig(ctx context.Context, kubeclient client.Client, instrumentationConfig *odigosv1.InstrumentationConfig, newStatus odigosv1.InstrumentationConfigStatus) error {
+	// This come to make sure we're updating instrumentationConfig only once (at the first time)
+	currentConfig := &odigosv1.InstrumentationConfig{}
+	err := kubeclient.Get(ctx, client.ObjectKeyFromObject(instrumentationConfig), currentConfig)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve current InstrumentationConfig: %w", err)
+	}
+
+	// Check if status is already set
+	if len(currentConfig.Status.RuntimeDetailsByContainer) != 0 {
```

Comment:
Why not update always? This means only the first Odiglet that runs this wins, so we will not update the value if the user changed the docker image?

Reply:
correct, it wont work with cases where user changed the value exists in Dockerfile.
we design this solution with this gap in mind